chain/chain/src/resharding/event_type.rs

```diff
@@ -26,34 +26,35 @@ pub struct ReshardingSplitShardParams {
     pub right_child_shard: ShardUId,
     /// The account at the boundary between the two children.
     pub boundary_account: AccountId,
-    /// Hash of the first block having the new shard layout.
+    /// Hash of the last block having the old shard layout.
     pub block_hash: CryptoHash,
     /// The block before `block_hash`.
     pub prev_block_hash: CryptoHash,
 }
 
 impl ReshardingEventType {
-    /// Takes as input a [ShardLayout] definition and deduces which kind of resharding operation must be
-    /// performed.
+    /// Takes as input a [ShardLayout] definition and deduces which kind of resharding operation
+    /// must be performed.
     ///
     /// # Args:
-    /// * `shard_layout`: the new shard layout
-    /// * `block_hash`: hash of the first block with `shard_layout`
+    /// * `next_shard_layout`: the new shard layout
+    /// * `block_hash`: hash of the last block with the shard layout before `next_shard_layout`
     /// * `prev_block_hash`: hash of the block preceding `block_hash`
     ///
-    /// Returns a [ReshardingEventType] if exactly one resharding change is contained in `shard_layout`, otherwise returns `None`.
+    /// Returns a [ReshardingEventType] if exactly one resharding change is contained in
+    /// `next_shard_layout`, otherwise returns `None`.
```

Comment:
Is it actually possible for this method to return None? 

It sounds a bit like if there is non-1 number of events None would be returned but it actually returns an error right?

Reply:
The method will return none if there's no resharding at all in the shard layout, like when all children are child of themselves. It's tested in `parse_event_type_from_shard_layout`